dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java

```diff
@@ -1775,16 +1775,24 @@ public boolean deleteAsset(String inode) throws Exception
             throw new DotRuntimeException ("Error trying to obtain the current liferay user from the request.");
         }
 
-    	HashMap<String, Object> result = new HashMap<String, Object> ();
-    	Inode asset = (Inode) InodeFactory.getInode(inode, Inode.class);
-    	if (asset instanceof Folder) {
-    		Folder folder = (Folder) asset;
+    	final Map<String, Object> result = new HashMap<> ();
+		Folder folder = null;
+		try {
+			folder = APILocator.getFolderAPI().find(inode, user, false);
+		} catch (DotSecurityException e) {
+			Logger.error(this, "Error trying to get info for folder with inode: " + inode, e);
+			throw new DotRuntimeException ("Error changing asset menu order.");
+		}
+
+		if (null != folder) {
     		result.put("lastValue", folder.getSortOrder());
-    		WebAssetFactory.changeAssetMenuOrder (asset, newValue, user);
+    		WebAssetFactory.changeAssetMenuOrder (folder, newValue, user);
     	} else {
+			Inode asset = InodeFactory.getInode(inode, Inode.class);
```

Comment:
since you are using the factory here, you should annotated the method with close db

Reply:
Do we use this annotation in Ajax classes? I didn't include this logic, this code is legacy. I just added a validation for folders

Reply:
In this particular case, it is right